src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -2109,8 +2107,16 @@ JRT_LEAF(void, SharedRuntime::fixup_callers_callsite(Method* method, address cal
   // "to interpreter" stub in order to load up the Method*. Don't
   // ask me how I know this...
 
+  // Result from nmethod::is_unloading is not stable across safepoints.
+  NoSafepointVerifier nsv;
+
+  CompiledMethod* callee = moop->code();
+  if (callee == NULL) {
+    return;
+  }
+
   CodeBlob* cb = CodeCache::find_blob(caller_pc);
-  if (cb == NULL || !cb->is_compiled() || entry_point == moop->get_c2i_entry()) {
+  if (cb == NULL || !cb->is_compiled() || callee->is_unloading()) {
```

Comment:
Why not move the is_unloading check on callee to the if statement just above that checks the callee (as opposed to the callsite)?

Reply:
I guess I was thinking is_unloading() can be a bit expensive the first time it is called, so it might be better to fail for other reasons first.  But I believe is_unloading will eventually be called for every nmethod each unloading cycle, so avoiding the cost here just means moving it to somewhere else.  I can move it to where you suggest if you like.

Reply:
Okay I see. I'll leave it to you to decide if you prefer to move it or not. I'm okay with it either way.